src/PimApi/PimCatalogApiClient.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\PimApi;
+
+use App\PimApi\Model\Catalog;
+use App\Storage\AccessTokenStorageInterface;
+use App\Storage\PimURLStorageInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class PimCatalogApiClient
+{
+    public function __construct(
+        private HttpClientInterface $client,
+        private readonly PimURLStorageInterface $pimURLStorage,
+        AccessTokenStorageInterface $accessTokenStorage,
+    ) {
+        $this->client = $this->client->withOptions([
+            'headers' => [
+                'Content-type' => 'application/json',
+                'Authorization' => 'Bearer '.$accessTokenStorage->getAccessToken(),
```

Comment:
```suggestion
                'Authorization' => 'Bearer '.$accessTokenStorage->getAccessToken(),
```

Same, it's nullable, I would rather have a clean LogicException logged in datadog instead of a cryptic error "Access Denied"